syncer/syncer.go

```diff
@@ -218,11 +219,18 @@ func (s *syncer) bulkSyncWithPeer(peerID peer.ID, newBlockCallback func(*types.F
 		return 0, false, err
 	}
 
+	// Create a blockchain subscription for the sync progression and start tracking
+	s.syncProgression.StartProgression(localLatest+1, s.blockchain.SubscribeEvents())
+	s.syncProgression.UpdateHighestProgression(peerLatestBlock)
+
 	defer func() {
 		err := s.syncPeerClient.CloseStream(peerID)
 		if err != nil {
 			s.logger.Error("Failed to close stream: ", err)
 		}
+
+		// Stop monitoring the sync progression upon exit
+		s.syncProgression.StopProgression()
 	}()
 
```

Comment:
IMO we should expose

`UpdateCurrentProgression(currentBlock uint64)` function to `Progression` interface and invoke it whenever `blockCh` gets read, e.g. here, after this line (https://github.com/0xPolygon/polygon-edge/blob/2351bc8597b0c6a127074cd17f345d1a0c57b6f5/syncer/syncer.go#L266). WDYT?

Reply:
Sorry, I missed that out. I was looking into restore functionality and there they are explicitly invoking `UpdateCurrentProgression`.